dkpro-core-io-brat-asl/src/main/java/de/tudarmstadt/ukp/dkpro/core/io/brat/internal/model/BratTextAnnotation.java

```diff
@@ -29,84 +29,116 @@
     private static final Pattern PATTERN = Pattern.compile(
             "(?<ID>T[0-9]+)\\t" + 
             "(?<TYPE>[a-zA-Z0-9_][a-zA-Z0-9_\\-]+) " +
-            "(?<BEGIN>[0-9]+) " +
-            "(?<END>[0-9]+)" +
-            "(;[0-9]+ [0-9]+)*\\t" +
+            "(?<OFFSET>[0-9]+ [0-9]+(;[0-9]+ [0-9]+)*)\\t" +
             "(?<TEXT>.*)");
     
     private static final String ID = "ID";
     private static final String TYPE = "TYPE";
-    private static final String BEGIN = "BEGIN";
+    private static final String OFFSET = "OFFSET";
     private static final String END = "END";
     private static final String TEXT = "TEXT";
     
-    private final int begin;
-    private final int end;
-    private final String text;
+    private final int[] begin;
+    private final int[] end;
+    private final String[] text;
 
-    public BratTextAnnotation(int aId, String aType, int aBegin, int aEnd, String aText)
+    public BratTextAnnotation(int aId, String aType, int[] aBegin, int[] aEnd, String[] aText)
     {
         this("T" + aId, aType, aBegin, aEnd, aText);
     }
 
-    public BratTextAnnotation(String aId, String aType, int aBegin, int aEnd, String aText)
+    public BratTextAnnotation(String aId, String aType, int[] aBegin, int[] aEnd, String[] aText)
     {
         super(aId, aType);
         begin = aBegin;
         end = aEnd;
         text = aText;
     }
 
-    public int getBegin()
+    public BratTextAnnotation(String aId, String aType, String aOffset, String aText)
+    {
+        super(aId, aType);
+        BratTextOffset bratTextOffset = new BratTextOffset(aOffset);
+        begin = bratTextOffset.getBegin();
+        end = bratTextOffset.getEnd();
+        text = splitText(aText, begin, end);
```

Comment:
I'd rather have a constructor `(String aId, String aType, List<Offsets> aOffsets, String[] aText)` and handle all the parsing code in the `parse` method - including splitting the text.